addons/l10n_in_withholding/wizard/l10n_in_account_withhold_wizard.py

```diff
@@ -0,0 +1,303 @@
+            if wizard.withhold_type in ('in_withhold', 'in_refund_withhold'):
+                l10n_in_tds_tax_type = 'purchase'
+            elif wizard.withhold_type in ('out_withhold', 'out_refund_withhold'):
+                l10n_in_tds_tax_type = 'sale'
+            wizard.l10n_in_tds_tax_type = l10n_in_tds_tax_type
+
+    @api.depends('related_move_id', 'related_payment_id')
+    def _compute_partner_id(self):
+        for wizard in self:
+            wizard.partner_id = wizard.related_move_id.partner_id or wizard.related_payment_id.partner_id
+
+    @api.depends('related_move_id', 'related_payment_id')
+    def _compute_company_id(self):
+        for wizard in self:
+            wizard.company_id = wizard.related_move_id.company_id or wizard.related_payment_id.company_id
+
+    @api.depends('company_id')
+    def _compute_journal(self):
+        for wizard in self:
+            wizard.journal_id = wizard.company_id.l10n_in_withholding_journal_id or \
+                                wizard.env['account.journal'].search([('company_id', '=', wizard.company_id.id), ('type', '=', 'general')], limit=1)
+
+    @api.depends('related_move_id', 'related_payment_id')
+    def _compute_withhold_type(self):
+        for wizard in self:
+            if wizard.related_move_id:
+                move_type = wizard.related_move_id.move_type
+                wizard.withhold_type = {
+                    'out_invoice': 'out_withhold',
+                    'in_invoice': 'in_withhold',
+                    'out_refund': 'out_refund_withhold',
+                    'in_refund': 'in_refund_withhold',
+                }[move_type]
+            else:
+                wizard.withhold_type = 'in_withhold' if wizard.related_payment_id.payment_type == 'outbound' else 'out_withhold'
+
+    @api.depends('related_move_id', 'related_payment_id', 'withhold_line_ids.base')
+    def _compute_display_amount_warning(self):
+        for wizard in self:
+            if wizard.related_move_id:
+                wizard.display_amount_warning = wizard.related_move_id.amount_untaxed < sum(line.base for line in wizard.withhold_line_ids)
+            else:
+                wizard.display_amount_warning = wizard.related_payment_id.amount < sum(line.base for line in wizard.withhold_line_ids)
+
+    # ===== MOVE CREATION METHODS =====
+    def action_create_and_post_withhold(self):
+        self.ensure_one()
+        withholding_account_id = self.company_id.l10n_in_withholding_account_id
+        self._validate_withhold_data_on_post(withholding_account_id)
+
+        # Withhold creation and posting
+        vals = self._prepare_withhold_header()
+        move_lines = self._prepare_withhold_move_lines(withholding_account_id)
+        vals['line_ids'] = [Command.create(line) for line in move_lines]
+        withhold = self.env['account.move'].create(vals)
+        withhold.action_post()
+
+        # If the withhold is created from a payment, there is no need to reconcile
```

Comment:
Because it will be reconciled later with the invoice as well?  (As we should be talking about advance payments?  Should we avoid the TDS button on the payment if the payment has been reconciled already?)

Reply:
I will hide the TDS button on the payment if it has been reconciled.